appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/JaxrsContainerRequestTracingFilter.java

```diff
@@ -125,11 +130,11 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
                 tracedAnnotation = OpenTracingCdiUtils.getAnnotation(beanManager, Traced.class, resourceInfo);
             }
             
-            // If there is no annotation, or if there is an annotation and a config override indicating that we should
-            // trace the method...
-            if (tracedAnnotation == null || (boolean) OpenTracingCdiUtils.getConfigOverrideValue(
-                    Traced.class, "value", resourceInfo, boolean.class)
-                    .orElse(tracedAnnotation.value())) {
+            // If there is no matching skip pattern and no traced annotation, or if there is there is no matching skip
+            // pattern and a traced annotation set to true (via annotation or config override)
```

Comment:
This method is quite complex. Could use the logical seperation by the comments as points for extracting methods. Just an observation, not a request to change now.

Reply:
In reference to the if block? Or the method itself? Because if the latter it's not that complex? Everything above the if is just variable initialisation.